src/pages/index.js

```diff
@@ -2,97 +2,224 @@ import "./index.css"; // add import of the main stylesheets file
-});
+const confirmationModal = new PopupWithConfirmation(
+  selectors.confirmationModalSelector
+);
+confirmationModal.setEventListeners();
+
+Promise.all([api.getUserInfo(), api.getInitialCards()])
+  .then(([data, cards]) => {
+    userInfo.setUserInfo({
+      name: data.name,
+      about: data.about,
+    });
+    userInfo.setAvatar(data.avatar);
+
+    section = new Section(
+      {
+        items: cards,
+        renderer: (cardData) => {
+          section.addItem(renderCard(cardData, userId));
+        },
+      },
+      selectors.cardListSelector
+    );
+    section.renderItems();
+  })
+  .catch((err) => {
+    console.error(err);
+  });
 
-const previewImagePopup = new PopupWithImage(previewImageModal);
+const formValidators = {};
+
+const enableValidation = (validationSettings) => {
+  const formList = Array.from(
+    document.querySelectorAll(validationSettings.formSelector)
+  );
+  formList.forEach((formElem) => {
+    const validator = new FormValidator(validationSettings, formElem);
+    const formName = formElem.getAttribute("name");
+    formValidators[formName] = validator;
+    validator.enableValidation();
+  });
+};
+enableValidation(validationSettings);
 
-// Event Handlers
+editProfileButton.addEventListener("click", () => {
+  const info = userInfo.getUserInfo();
 
-function handleEditProfileButton() {
-  editProfileFormPopup.open();
-  editProfileFormPopup.setInputValues(userInfo.getUserInfo());
-  editProfileFormValidator.hideModalInputError();
-  editProfileFormValidator.resetButtonState();
-}
+  profileNameInput.value = info.name;
+  profileDescriptionInput.value = info.about;
 
-function handleAddCardButton() {
-  addCardFormValidator.resetButtonState();
-  addCardFormValidator.hideModalInputError();
-  addCardFormPopup.open();
-}
+  profileModal.open();
+  formValidators["profile-form"].resetValidation();
+});
 
-// Event Listeners
+const renderCard = ({ name, link, _id, ownerId, isLiked }, userId) => {
+  const card = new Card({
+    name,
+    link,
+    userId,
+    _id,
+    ownerId,
+    isLiked,
+    handleImageClick,
+    handleDeleteClick,
+    handleLikeClick,
+    cardSelector: selectors.cardSelector,
+  });
 
-editProfileFormPopup.setEventListeners();
-editProfileButton.addEventListener("click", handleEditProfileButton);
-addCardFormPopup.setEventListeners();
-addCardButton.addEventListener("click", handleAddCardButton);
-previewImagePopup.setEventListeners();
+  return card.getView();
+};
+
+const handleImageClick = ({ name, link }) => {
+  imageModal.open({ name, link });
+};
+
+const avatarModal = new PopupWithForm(
+  selectors.avatarModalSelector,
+  (avatarData) => {
+    avatarModal.setButtonText(true);
+    api
+      .updateAvatar(avatarData)
+      .then((data) => {
+        userInfo.setAvatar(data.avatar);
+        avatarModal.close();
+      })
+      .catch((err) => {
+        console.error(err);
+      })
+      .finally(() => avatarModal.setButtonText());
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```